echidna/core/dsextract.py

```diff
@@ -0,0 +1,608 @@
+import numpy
+import rat
+import echidna.calc.constants as const
+
+
+def function_factory(dimension, **kwargs):
+    '''Factory function that returns a dsextract class
+    corresponding to the dimension (i.e. DS parameter)
+    that is being extracted.
+
+    Args:
+      dimension (str): to extract from a RAT DS/ntuple file.
+      \**kwargs (dict): to be passed to and checked by the extractor.
+
+    Raises:
+      IndexError: dimension is an unknown parameter
+
+    Retuns:
+      Extractor object.
```

Comment:
Could you fix the Returns section too so it gives the return type and description please.